app/src/main/java/org/simple/clinic/newentry/PatientEntryEffectHandler.kt

```diff
@@ -81,9 +81,18 @@ class PatientEntryEffectHandler @AssistedInject constructor(
         .addAction(OpenMedicalHistoryEntryScreen::class.java, ui::openMedicalHistoryEntryScreen, schedulersProvider.ui())
         .addTransformer(LoadInputFields::class.java, loadInputFields())
         .addConsumer(SetupUi::class.java, { ui.setupUi(it.inputFields) }, schedulersProvider.ui())
+        .addTransformer(FetchColonyOrVillagesEffect::class.java, fetchColonyOrVillages())
         .build()
   }
 
+  private fun fetchColonyOrVillages(): ObservableTransformer<FetchColonyOrVillagesEffect, PatientEntryEvent> {
+    var colonyOrVillages: List<String>? = null
+    return ObservableTransformer { fetchColonyOrVillagesEffect ->
+      fetchColonyOrVillagesEffect.doOnNext { colonyOrVillages = patientRepository.allColoniesOrVillagesInPatientAddress() }
+          .map { ColonyOrVillagesFetched(colonyOrVillages) }
+    }
```

Comment:
We should not be using a `var` variable here to pass the list information downstream. Instead, use `map`.

```kt
fetchColonyOrVillagesEffect
    .map { patientRepository.allColoniesOrVillagesInPatientAddress() }
    .map(::ColonyOrVillagesFetched)
```